app/views/handlers/feedback.py

```diff
@@ -86,29 +90,43 @@ def handle_post(self) -> None:
         session_data: SessionData = self._session_store.session_data  # type: ignore
         session_data.feedback_count += 1
 
-        feedback_metadata = FeedbackMetadata(self._questionnaire_store.metadata["case_id"], self._questionnaire_store.metadata["tx_id"])  # type: ignore
+        metadata = self._questionnaire_store.metadata
+        if not metadata:
+            raise NoMetadataException  # pragma: no cover
+
+        case_id = metadata.case_id
+        tx_id = metadata.tx_id
 
         # pylint: disable=no-member
         # wtforms Form parents are not discoverable in the 2.3.3 implementation
-        feedback_message = FeedbackPayload(
-            metadata=self._questionnaire_store.metadata,
+        feedback_converter = (
+            FeedbackPayloadV2
+            if metadata.version is AuthPayloadVersion.V2
+            else FeedbackPayload
+        )
+        feedback_message = feedback_converter(
+            metadata=metadata,
             response_metadata=self._questionnaire_store.response_metadata,
             schema=self._schema,
-            case_id=self._questionnaire_store.metadata["case_id"],
+            case_id=case_id,
             submission_language_code=session_data.language_code,
             feedback_count=session_data.feedback_count,
             feedback_text=self.form.data.get("feedback-text"),
             feedback_type=self.form.data.get("feedback-type"),
         )
-        message = feedback_message()
-        metadata = feedback_metadata()
-        message.update(metadata)
+
         encrypted_message = encrypt(
-            message, current_app.eq["key_store"], KEY_PURPOSE_SUBMISSION  # type: ignore
+            feedback_message(), current_app.eq["key_store"], KEY_PURPOSE_SUBMISSION  # type: ignore
+        )
+
+        additional_metadata = get_receipting_metadata(metadata)
+
+        feedback_metadata = FeedbackMetadata(
+            tx_id=tx_id, case_id=case_id, **additional_metadata
         )
 
         if not current_app.eq["feedback_submitter"].upload(  # type: ignore
-            metadata, encrypted_message
+            feedback_metadata, encrypted_message
```

Comment:
Feedback is currently broken. This need to be calling the instance to return a dict. The type ignore is hiding the error.
```suggestion
            feedback_metadata(), encrypted_message
```

Reply:
Updated 👍